api/src/main/java/com/lifegraph/team20/controllers/ApiSearchGraphController.java

```diff
@@ -2,44 +2,67 @@
 
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.sql.Timestamp;
 import java.util.List;
+import java.util.Optional;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.RowMapper;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.lifegraph.team20.models.SearchGraph;
 
 @RestController
 public class ApiSearchGraphController {
 
-	private static final String LikeName = "は";
+//	private static final String like_name = "";
+//	private static final String start_date = "2020-05-01";
+//	private static final String finish_date = "2020-06-01";
 
-	@RequestMapping(value = "/auth/search", method = RequestMethod.GET)
 //	メソッドや処理を関連づけるアノテーション
-	public List<SearchGraph>SearchGraphs(){
-		List<SearchGraph> SearchGraphs = SelectSearchGraph();
-//		l.29のselectSearchGraphを呼び出す
-		return SearchGraphs;
-	}
+	@GetMapping(value = "/auth/search")
+//	ResponseEntity<String> doPost(@RequestBody UploadForm body){
+		public  List<SearchGraph>SearchGraphs(@RequestParam("LikeName") Optional<String>like_name,
+				@RequestParam("StartDate") Optional<Timestamp>start_date,
+				@RequestParam("FinishDate") Optional<Timestamp>finish_date){
+//		Optional:その値がnullかもしれないことを表現するクラス
+//				URLが叩かれたときにこれが動く
+			List<SearchGraph> SearchGraphs = SelectSearchGraph(like_name,start_date,finish_date);
+//				l.29のselectSearchGraphを呼び出す
+			return SearchGraphs;
+		}
 
+//	}
   @Autowired
   //MySQLのデータを持ってくるライブラリ
   private JdbcTemplate jdbcTemplate;
-	private List<SearchGraph> SelectSearchGraph(){
-		final String sql = "select name,user_id,updated_at from users inner join parent_graphs on users.id "
-				+ "= parent_graphs.user_id  where name like '%"+LikeName+"%'";
-//		sqlに"select ~"という文字列をいれる
+	private List<SearchGraph> SelectSearchGraph(Optional<String> like_name,Optional<Timestamp> start_date,Optional<Timestamp> finish_date){
+		String sql="";
+//		ここにif文をいれる 名前検索が入ったらと日時検索が入ったら
+		if(like_name != null) {
```

Comment:
```suggestion
		if(likeName.isPresent()) {
```
Optionalを利用しているので、こういう書き方をします。
そのままの書き方だと、likeNameはnullにならないと思われる

Reply:
修正しました。